components/bl-treeView-component/src/index.js

```diff
@@ -0,0 +1,160 @@
+    handlerItemClick(value, label);
+  }, []);
+
+  const handlerItemClick = useCallback((value, label) => {
+    onClick({ label, value });
+    setSelectedItemId(value);
+  }, []);
+
+  if (!display) {
+    return null;
+  }
+
+  return (
+    <div className={ cn('bl-customComponent-treeView', classList) } style={ style }>
+      <Branch
+        itemsTree={ itemsTree }
+        gap={ gap }
+        parentItems={ parentItems }
+        selectedItemId={ selectedItemId }
+        openHandler={ openHandler }
+        handlerItemClick={ handlerItemClick }
+      />
+    </div>
+  );
+}
+
+function Branch(props) {
+  const { itemsTree, parentItems, gap, selectedItemId, handlerItemClick, openHandler } = props;
+
+  return (
+    <ul className="list">
+      { itemsTree.map(item => {
+        const { value, label, children, levelOfNesting } = item;
+        const nestingStyle = { marginLeft: gap * levelOfNesting + 'px' };
+
+        if (children) {
+          const { isOpen } = parentItems.find(element => element.value === value);
+
+          return (
+            <li className="container">
+              <ParentItem
+                selectedItemId={ selectedItemId }
+                openHandler={ openHandler }
+                nestingStyle={ nestingStyle }
+                value={ value }
+                label={ label }
+                isOpen={ isOpen }
+              />
+
+              { isOpen && (
+                <Branch
+                  itemsTree={ children }
+                  gap={ gap }
+                  parentItems={ parentItems }
+                  selectedItemId={ selectedItemId }
+                  openHandler={ openHandler }
+                  handlerItemClick={ handlerItemClick }
+                />
+              ) }
+            </li>
+          );
+        }
+
+        return (
+          <Item
+            selectedItemId={ selectedItemId }
+            handlerItemClick={ handlerItemClick }
+            nestingStyle={ nestingStyle }
+            value={ value }
+            label={ label }
+          />
+        );
+      }) }
+    </ul>
+  );
+}
+
+function Item({ selectedItemId, handlerItemClick, nestingStyle, value, label }) {
+  return (
+    <li
+      tabIndex={ selectedItemId === value ? 0 : -1 }
+      className={ cn('list__item', { selected: selectedItemId === value }) }
+      onClick={ () => handlerItemClick(value, label) }
+      style={ nestingStyle }>
+      { label }
+    </li>
+  );
+}
+
+function ParentItem({ selectedItemId, openHandler, nestingStyle, value, label, isOpen }) {
+  return (
+    <div
+      tabIndex={ selectedItemId === value ? 0 : -1 }
+      className={ cn('list__button', { selected: selectedItemId === value, open: isOpen }) }
+      onClick={ () => openHandler(value, label) }
+      style={ nestingStyle }>
+      <ButtonIcon/>
+      { label }
+    </div>
+  );
+}
+
+function ButtonIcon() {
```

Comment:
`ButtonIcon` is a general name, like we can pass there any icon, 
need to rename it to a more specific